src/test/isolation2/input/resgroup/resgroup_move_query.source

```diff
@@ -141,9 +151,134 @@ SELECT is_session_in_group(pid, 'rg_move_query_mem_small') FROM pg_stat_activity
+1: BEGIN;
+1&: SELECT pg_sleep(5) FROM gp_dist_random('gp_id');
+2: SELECT gp_inject_fault('resource_group_give_away2-1', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-1', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-2', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away2-2', 'error', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2&: SELECT gp_toolkit.pg_resgroup_move_query(pid, 'rg_move_query') FROM pg_stat_activity WHERE query LIKE '%pg_sleep%' AND rsgname='rg_move_query_mem_small';
+3: SELECT pg_sleep(3);
+3: SELECT gp_inject_fault('resource_group_give_away2-1', 'resume', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2<:
+2: SELECT gp_inject_fault('resource_group_move_handler2', 'resume', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+1<:
+--is_session_in_group works only if all backends moved to group, so it will show 'f', but gp_resgroup_status will show actual result
+2: SELECT is_session_in_group(pid, 'rg_move_query') FROM pg_stat_activity WHERE query LIKE '%pg_sleep%' AND state = 'idle in transaction';
+2: SELECT num_running FROM gp_toolkit.gp_resgroup_status WHERE rsgname='rg_move_query';
+1: END;
+2: SELECT num_running FROM gp_toolkit.gp_resgroup_status WHERE rsgname='rg_move_query';
+1q:
+
+-- test11: check destination group has no slot leaking if target process set latch at the last moment
+-- sleep at first process
+-- start to move it at second process, send moving signal to first process
+-- suspend at first process just before setting latch
+-- wait for timeout on WaitLatch of second process and suspend
+-- resume at first process, it should set latch (which is late) and clean moveto* values
+-- resume at second process, as moveto* was cleaned, we know first process handled signal
+-- moving command at second process should finish successfully
+1: SET ROLE role_move_query_mem_small;
+1: SELECT gp_inject_fault('resource_group_move_handler1', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+1: SELECT gp_inject_fault('resource_group_move_handler1', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+1: BEGIN;
+1&: SELECT pg_sleep(3) FROM gp_dist_random('gp_id');
+2: SELECT gp_inject_fault('resource_group_give_away3', 'reset', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2: SELECT gp_inject_fault('resource_group_give_away3', 'suspend', dbid) FROM gp_segment_configuration where role = 'p' and content = -1;
+2&: SELECT gp_toolkit.pg_resgroup_move_query(pid, 'rg_move_query') FROM pg_stat_activity WHERE query LIKE '%pg_sleep%' AND rsgname='rg_move_query_mem_small';
+3: SELECT pg_sleep(4);
```

Comment:
Why do you expect that the 2nd session already achieve fault injector? Do we decrease default timeout from 30 seconds?

Reply:
I expect you are right.